src/ripple/app/tx/impl/Transactor.h

```diff
@@ -158,7 +176,60 @@ class Transactor
         uint256 const& ticketIndex,
         beast::Journal j);
 
+
+    // Hooks
+
+    static FeeUnit64
+    calculateHookChainFee(ReadView const& view, STTx const& tx, Keylet const& hookKeylet,
+            bool collectCallsOnly = false);
+
 protected:
+
+    void
+    doHookCallback(
+        std::shared_ptr<STObject const> const& provisionalMeta);
+
+    TER
+    doTSH(
+        bool strong,                                // only do strong TSH iff true, otheriwse only weak
+        hook::HookStateMap& stateMap,
+        std::vector<hook::HookResult>& result,
+        std::shared_ptr<STObject const> const& provisionalMeta);
+
+
+    // Execute a hook "Again As Weak" is a feature that allows
+    // a hook that which is being executed pre-application of the otxn
+    // to request an additional post-application execution.
+    void
+    doAaw(
```

Comment:
Ah, so that's what the name means. I think `doAgainAsWeak` is a fine name.

Reply:
👍